lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+            specimen_observation_entry,
+            specimen_entry_v2,
+            specimen_observation_entry_v2,
+            weekly_encounter_entry,
+            weekly_questionnaire_entry,
+            weekly_computed_questionnaire_entry
+        ]
+
+        persisted_resource_entries.extend(list(filter(None, current_instance_entries)))
+
+    return create_bundle_resource(
+        bundle_id = str(uuid4()),
+        timestamp = datetime.now().astimezone().isoformat(),
+        source = f"{REDCAP_URL}{enrollment.project.id}/{enrollment.id}",
+        entries = list(filter(None, persisted_resource_entries))
+    )
+
+
+def airs_get_encounter_date(record: REDCapRecord, event_type: EventType) -> Optional[str]:
+    encounter_date = None
+
+    # TODO: I don't think we need to proceed if wk_date isn't defined
+    #  (ie, this weekly encounter isn't complete yet)
+    if event_type == EventType.ENCOUNTER:
+        encounter_date = extract_date_from_survey_timestamp(record, 'weekly') \
+
+        # This will never be true for AIRS since 'daily_attestation' isn't a thing
+        if encounter_date is None and is_complete('daily_attestation', record) and record.repeat_instance:
+            encounter_date = get_date_from_repeat_instance(record.repeat_instance)
+
+    elif event_type == EventType.ENROLLMENT:
+        encounter_date = record.get('enr_date_complete') and \
+            datetime.strptime(record.get('enr_date_complete'), '%Y-%m-%d').strftime('%Y-%m-%d')
+
+    return encounter_date
+
+
+def create_encounter_id(record: REDCapRecord, is_followup_encounter: bool) -> str:
+    """
+    Create the encounter_id from the REDCap *record*.
+    """
+    if record.event_name:
+        redcap_event_name = record.event_name
+    else:
+        redcap_event_name = ''
+
+    if record.repeat_instance:
+        redcap_repeat_instance = str(record.repeat_instance)
+    else:
+        redcap_repeat_instance = ''
+
+    if is_followup_encounter:
+        encounter_identifier_suffix = "_follow_up"
+    else:
+        encounter_identifier_suffix = ''
+
+    return f'{record.project.base_url}{record.project.id}/{record.id}/{redcap_event_name}/' + \
+        f'{redcap_repeat_instance}{encounter_identifier_suffix}'
+
+
+def get_collection_date(record: REDCapRecord, collection_method: CollectionMethod) -> Optional[str]:
+    """
+    Determine sample/specimen collection date from the given REDCap *record*.
+    """
+    # For all surveys, try the survey _timestamp field (which is in Pacific time)
+    # before custom fields because the custom fields aren't always populated and when
+    # they are populated they use the browser's time zone.
+    collection_date = None
+
+    if collection_method == CollectionMethod.KIOSK:
+        collection_date = extract_date_from_survey_timestamp(record, "kiosk_registration_4c7f") or record.get("nasal_swab_q")
+
+    elif collection_method in [CollectionMethod.SWAB_AND_SEND, CollectionMethod.UW_DROPBOX]:
```

Comment:
For swab and send in this study, I think the redcap instruments you'll want to check for the collection date are airs_kit_activation and airs_kit_activation_2 . Kiosk and dropbox won't be relevant for airs.

Reply:
Yes, this was left over from the UW reopening code. Removed.